src/main/java/seedu/address/storage/JsonSerializableTutorsPet.java

```diff
@@ -68,11 +74,27 @@ private void studentsToModelType(TutorsPet tutorsPet) throws IllegalValueExcepti
      * @throws IllegalValueException if there were any data constraints violated.
      */
     private void classesToModelType(TutorsPet tutorsPet) throws IllegalValueException {
+        // Get all UUIDs under "students" field in tutorspet.json.
+        ObservableList<Student> students = tutorsPet.getStudentList();
+        Set<UUID> uniqueStudentUuids = new HashSet<>();
+        for (Student student : students) {
+            uniqueStudentUuids.add(student.getUuid());
+        }
+
         for (JsonAdaptedModuleClass jsonAdaptedModuleClass : classes) {
             ModuleClass moduleClass = jsonAdaptedModuleClass.toModelType();
             if (tutorsPet.hasModuleClass(moduleClass)) {
                 throw new IllegalValueException(MESSAGE_DUPLICATE_MODULE_CLASS);
             }
+
+            // Check that the set of student UUIDs within a class is a subset of that of
+            // all student UUIDs in uniqueStudentUuids. Otherwise something is wrong and
+            // Tutor's Pet will not boot up.
```

Comment:
"something is wrong" feels a little generic.. Perhaphs it is good to specify?
```suggestion
            // Check that the set of student UUIDs within a class is a subset of that of
            // all student UUIDs in uniqueStudentUuids. Otherwise, Tutor's Pet would not
            // boot up due to data corruption.
```